src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -3732,48 +3742,139 @@ public enum DarknessAblaze implements LogicCardInfo {
       return itemCard (this) {
         text "Search your deck for a Pokémon with the same name as a Pokémon in your discard pile, and add it to your hand. Then, shuffle your deck."
         onPlay {
+          def pokeNamesInDiscard = []
+          my.discard.filterByType(POKEMON).each{
+            pokeNamesInDiscard.add(it.name)
+          }
+          my.deck.search("Pokémon with same name as a Pokémon in your discard pile. Possible options: $pokeNamesInDiscard", {it.cardTypes.is(POKEMON) && pokeNamesInDiscard.contains(it.name)}).moveTo(my.hand)
         }
         playRequirement{
+          my.deck : "You have no cards in your deck"
+          my.discard.filterByType(POKEMON) : "You have no Pokémon in your discard pile"
         }
       };
       case PIERS_173:
       return supporter (this) {
-        text "Search your deck for 1 [D] Pokémon and any 1 Energy, reveal them, and put them into your hand. Then, shuffle your deck."
-        onPlay {
-        }
-        playRequirement{
-        }
+        text "Search your deck for a [D] Pokémon and an Energy card, reveal them, and put them into your hand. Then, shuffle your deck."
+          onPlay {
+            my.deck.search(count: 1, "Search your deck for a [D] Pokémon",pokemonTypeFilter(D)).moveTo(my.hand)
+            my.deck.search("Search your deck for an Energy card",cardTypeFilter(ENERGY)).moveTo(my.hand)
+            shuffleDeck()
+          }
+          playRequirement{
+            assert my.deck.notEmpty
+          }
       };
       case POKEMON_BREEDER_S_NURTURING_174:
       return supporter (this) {
         text "Choose up to 2 of your Pokémon in play. Search your deck for a card that evolves from each of those Pokémon and put it on top of the Pokémon to evolve it. Then, shuffle your deck. (You can’t use this card during your first turn or on a Pokémon that just came into play.)"
         onPlay {
+          def pcsList = my.all.findAll {it.turnCount < bg.turnCount}
+          if (pcsList.size() > 1) {
+            pcsList = pcsList.select(max:2, "Choose up to 2 of your Pokémon in play that can make a regular evolve during this turn.")
+          }
+
+          pcsList.each{ pcs =>
+            def sel = deck.search ("Select a Pokémon that evolves from $pcs.name.", {
+              it.cardTypes.is(EVOLUTION) && it.predecessor == pcs.name
+            })
+            if (sel) { evolve(pcs, sel.first(), OTHER) }
+          }
+
+          shuffleDeck()
         }
         playRequirement{
+          assert my.deck : "Your deck is empty."
+          assert bg.turnCount > 2 : "Cannot use this card during your first turn."
+          assert my.all.findAll {it.turnCount < bg.turnCount}
```

Comment:
Would be nice to mention that it isn't usable on Pokémon put in play this turn here I think. Something like
```suggestion
          assert my.all.findAll {it.turnCount < bg.turnCount} : "Can't be used on a Pokémon brought in this turn"
```